packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,678 @@
+    const policyStatements = [
+      new iam.PolicyStatement({
+        resources: [
+          cdk.Stack.of(this).formatArn({
+            arnFormat: cdk.ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME,
+            service: 'emr-containers',
+            resource: 'virtualclusters',
+            resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.virtualClusterId.value) ? '*' : this.props.virtualClusterId.value, // Need wild card for dynamic start job run https://docs.aws.amazon.com/step-functions/latest/dg/emr-eks-iam.html
+          }),
+        ],
+        actions: ['emr-containers:StartJobRun'],
+        conditions: {
+          StringEquals: {
+            'emr-containers:ExecutionRoleArn': this.role.roleArn,
+          },
+        },
+      }),
+    ];
+
+    if (this.integrationPattern === sfn.IntegrationPattern.RUN_JOB) {
+      policyStatements.push(
+        new iam.PolicyStatement({
+          resources: [
+            cdk.Stack.of(this).formatArn({
+              arnFormat: cdk.ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME,
+              service: 'emr-containers',
+              resource: 'virtualclusters',
+              resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.virtualClusterId.value) ? '*' : `${this.props.virtualClusterId.value}/jobruns/*`, // Need wild card for dynamic start job run https://docs.aws.amazon.com/step-functions/latest/dg/emr-eks-iam.html
+            }),
+          ],
+          actions: [
+            'emr-containers:DescribeJobRun',
+            'emr-containers:CancelJobRun',
+          ],
+        }),
+      );
+    }
+
+    return policyStatements;
+  }
+}
+
+/**
+ * The information about job driver for Spark submit.
+ */
+export interface SparkSubmitJobDriver {
+
+  /**
+   * The entry point of job application.
+   *
+   * Length Constraints: Minimum length of 1. Maximum length of 256.
+   */
+  readonly entryPoint: sfn.TaskInput;
+
+  /**
+   * The arguments for a job application in a task input object containing an array of strings
+   *
+   * Length Constraints: Minimum length of 1. Maximum length of 10280.
+   * @type string[]
+   *
+   * @default - No arguments defined
+   */
+  readonly entryPointArguments?: sfn.TaskInput;
+
+  /**
+   * The Spark submit parameters that are used for job runs.
+   *
+   * Length Constraints: Minimum length of 1. Maximum length of 102400.
+   *
+   * @default - No spark submit parameters
+   */
+  readonly sparkSubmitParameters?: string;
+}
+
+/**
+ * Specify the driver that the EMR Containers job runs on.
+ * The job driver is used to provide an input for the job that will be run.
+ */
+export interface JobDriver {
+
+  /**
+   * The job driver parameters specified for spark submit.
+   *
+   * @see https://docs.aws.amazon.com/emr-on-eks/latest/APIReference/API_SparkSubmitJobDriver.html
+   *
+   * @default - No spark submit job driver parameters specified.
+   */
+  readonly sparkSubmitJobDriver?: SparkSubmitJobDriver;
```

Comment:
we have no tests where this is undefined. does this mean it's actually required, or do you need to add a test for this case?

Reply:
Need to add test for when this is undefined

Reply:
Nope – if they want to no-op then they can add a Choice state. This should be made required